### PR TITLE
[WEB-1598] Adds Bias Check Filter

### PR DESCRIPTION
**Pull request by pugazhendhi-m**
## Greptile Summary

Introduces a new Bias Check filter pipeline that analyzes content for potential bias using the d4data/bias-detection-model, implementing both inlet and outlet filtering capabilities.

- Inconsistent bias labels between inlet ("Biased") and outlet ("LABEL_1") in `/pipelines/bias_check_pipeline.py` need alignment
- Missing error handling for model loading/cleanup in `on_startup()` and `on_shutdown()` methods
- Empty `/pipelines/bias_check_pipeline/valves.json` needs configuration for pipelines, priority, and threshold
- Warning prefix in outlet method needs proper spacing before content concatenation
- Missing version pins for TensorFlow dependencies in `requirements.txt`

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -797,7 +797,8 @@ async def perform_filters(request: PerformFiltersRequest):
         
         # Map of filter names to their pipeline IDs
         filter_map = {
-            "NSFW Filter": "nsfw_filter_pipeline"
```
Comment (vigneshsubbiah16) [inline main.py]: not sure if we need this. please confirm

File: pipelines/bias_check_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                break
+
+        if assistant_message:
+            content = assistant_message.get("content", "")
+            result = self.bias_model(content)[0]
+            bias_score = result["score"]
+            is_biased = result["label"] == "LABEL_1"
+
+            if is_biased and bias_score > self.valves.threshold:
+                warning_prefix = "⚠️ **Bias Warning:** This response may contain biased content.\n\n"
```
Comment (vigneshsubbiah16) [inline pipelines/bias_check_pipeline.py]: are we going to add the bias with prefix or going to error the response, and stop returning anything? if this is from the hallucination pipeline,  the hallucination pipeline is outdated and can be removed. idea way is to let the gateway know taht the response is biased, and the gateway can return both the LLM response and the error code with message about the potential bias in the response. The invoker can choose to do what they want about it.

File: requirements.txt [hunk 3]
```diff
@@ -64,3 +64,6 @@ beautifulsoup4
 # Llama Index for RAG
 llama-index
 llama-index-llms-ollama
+
+tf-keras
```
Comment (vigneshsubbiah16) [inline requirements.txt:68]: are we using this?

Comment (pugazhendhi-m) [reply requirements.txt:68]: Yeah. It was a dependency

File: main.py (hunk 1 above)
Comment (pugazhendhi-m) [reply main.py]: We have the enabled filters coming in the request body. We have to use that string to call the respective pipeline. So this mapping helps with that

**Review by gowshik450526511 — APPROVED** [review]

**Pull request merged**